### PR TITLE
Docs build cleanup

### LINKED ISSUE #534: Disable [source] links in search results
Example:

Searching for [color](https://docs.djangoproject.com/en/1.8/search/?q=color) gives the following result:
search_django_documentation_django_-_2015-10-07_23 38 24

Personally, I've never found those results to be particularly useful and I think it'd be better to simply remove them (rather than trying to score them lower in the search results).

From what I remember, we didn't have those before so I think it must have been a sphinx update that brought them on (and I'm fairly sure it's unrelated to the search refactor with elastic because I remember seeing those results before).

### PR DESCRIPTION
**Pull request by bmispelon**
Result of my Pycon CZ sprint with @HonzaKral.

This adds some tests and simplifies some of the logic in `update_docs`.

One practical result is that the elasticsearch indexing now needs to be done in a separate step by running the `update_index --delete` command.

This should fix #534.

### CODE REVIEW

File: requirements/common.txt [hunk 1]
```diff
@@ -5,7 +5,8 @@ django-flat-theme==0.9.3
 django-hosts==1.2
 django-push==0.7
 django-registration-redux==1.2
-elasticsearch-dsl==0.0.4
+elasticsearch-dsl==0.0.9
+elasticsearch>=1.0.0,<2.0.0
```
Comment (timgraham) [inline requirements/common.txt]: Why is this needed?

Comment (HonzaKral) [inline requirements/common.txt]: Elasticsearch (the server) 2.0 is slightly backwards incomaptible with 1.x currently used by the site. To mirror that there are also two versions of the python client - 1.x and 2.x. We just need to fix the correct one in place to correspond with the ES version we are using. Otherwise it would default to newest one.

**Comment by timgraham** [comment]
With the current deployment scripts, we don't have an easy way to run one management command right after another completes (if this is important). Maybe creating a new command that calls "update_docs --verbosity=0" (the current command) and "update_index --delete" is needed?

**Comment by bmispelon** [comment]
Are the deployment scripts available somewhere pulbic?

**Comment by bmispelon** [comment]
I pushed another commit with a new management command that combines the two. Would that work?

**Comment by bmispelon** [comment]
Turns out the new `update_docs` wasn't actually working (because of two issues with using `Path` objects instead of strings).

I fixed them, rebased and pushed (also tested the command locally).

**Comment by timgraham** [comment]
In my local testing:
```
$ python manage.py update_docs_and_index
...
  File "/home/tim/code/djangoproject.com/docs/management/commands/update_index.py", line 32, in handle
    self.log('Successfully indexed item %s' % id_)
  File "/home/tim/code/djangoproject.com/docs/management/commands/update_index.py", line 23, in log
    if self.verbosity >= level:
TypeError: unorderable types: int() >= str()
```
Just in case it matters, I noticed I have `elasticsearch 1.7.0` instead of the latest `1.X` which looks like `elasticsearch 1.9.0`.

File: docs/management/commands/update_docs_and_index.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
+from django.core.management import call_command
+from django.core.management.base import BaseCommand
+
+
+class Command(BaseCommand):
+    """
+    Update the docs then re-index them in elasticsearch.
```
Comment (timgraham) [inline docs/management/commands/update_docs_and_index.py]: reindex

File: docs/models.py [hunk 3]
```diff
@@ -117,6 +119,36 @@ def scm_url(self):
             url += '@stable/' + self.version + '.x'
         return url
 
+    @transaction.atomic
+    def sync_to_db(self, decoded_documents):
+        """
+        Sync the given list of documents (decoded fjson files from sphinx) to
+        the database. Deletes all the release's documents first then
+        re-inserts them as needed.
```
Comment (timgraham) [inline docs/models.py]: reinserts

**Comment by bmispelon** [comment]
The `update_index` command was actually mis-handling the `verbosity` argument.

That should be fixed now.
